src/hooks/useFocusTrap.ts

```diff
@@ -0,0 +1,46 @@
+// Based on https://hiddedevries.nl/en/blog/2017-01-29-using-javascript-to-trap-focus-in-an-element
```

Comment:
Question: Doesn't floating-ui package have a focusTrap? I though it did, but cannot find it.

Reply:
@ychhabra-eightfold can you help here?